src/main.js

```diff
@@ -55,23 +61,42 @@ if (crashReporterDirectory) {
 		}
 	}
 
-	// Crashes are stored in the temp directory by default, so we
+	// Crashes are stored in the crashDumps directory by default, so we
 	// need to change that directory to the provided one
-	console.log(`Found --crash-reporter-directory argument. Setting temp directory to be '${crashReporterDirectory}'`);
-	app.setPath('temp', crashReporterDirectory);
-
-	// Start crash reporter
-	const { crashReporter } = require('electron');
-	const productName = (product.crashReporter && product.crashReporter.productName) || product.nameShort;
-	const companyName = (product.crashReporter && product.crashReporter.companyName) || 'Microsoft';
-	crashReporter.start({
-		companyName: companyName,
-		productName: process.env['VSCODE_DEV'] ? `${productName} Dev` : productName,
-		submitURL: '',
-		uploadToServer: false
-	});
+	console.log(`Found --crash-reporter-directory argument. Setting crashDumps directory to be '${crashReporterDirectory}'`);
+	app.setPath('crashDumps', crashReporterDirectory);
+} else {
+	const appCenter = product.appCenter;
+	// If enable-crash-reporter argv is undefined then this is a fresh start,
+	// based on telemetry.enableCrashreporter settings, generate a UUID which
+	// will be used as crash reporter id and also update the json file.
+	if (argvConfig['enable-crash-reporter'] === undefined) {
+		process.argv.push('--create-crash-reporter-config');
+	}
+	// Disable Appcenter crash reporting if
+	// * --crash-reporter-directory is specified
+	// * enable-crash-reporter runtime argument is set to 'false'
+	// * --disable-crash-reporter command line parameter is set
+	if (appCenter && (argvConfig['enable-crash-reporter'] !== false) && !args['disable-crash-reporter']) {
+		const isWindows = (process.platform === 'win32');
+		const isLinux = (process.platform === 'linux');
+		const crashReporterId = argvConfig['crash-reporter-id'];
+		submitURL = isWindows ? appCenter[process.arch === 'ia32' ? 'win32-ia32' : 'win32-x64'] : isLinux ? appCenter[`linux-x64`] : appCenter.darwin;
+		submitURL = submitURL.concat('&uid=', crashReporterId, '&iid=', crashReporterId, '&sid=', crashReporterId);
+		// Send the id for child node process that are explicitly starting crash reporter.
+		// For vscode this is ExtensionHost process currently.
+		process.argv.push('--crash-reporter-id', crashReporterId);
+	}
 }
 
+// Start crash reporter for all processes
+crashReporter.start({
```

Comment:
I wonder how useful it is to start the crash reporter if the user has disabled it via settings?

Reply:
There is not harm in calling it irrespective of the settings, when the appcenter config is not set, it won't send any data to the server, adding this conditionally will only make this `if else` block less simple with no added benefits.

Reply:
Ok, just wanted to confirm.